### PR TITLE
Fixes theme change crashes

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -293,7 +294,12 @@ open class FenixApplication : LocaleAwareApplication() {
                 components.core.store,
                 onNewTabOverride = {
                     _, engineSession, url ->
-                        val session = Session(url, components.browsingModeManager.mode.isPrivate)
+                        val shouldCreatePrivateSession =
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt]: This is the alternative fix to messing with making the `browsingModeManager` a singleton on `components`. Ultimately we need to move towards that solution but there is a bit more we need to think about that we can do post code freeze

File: app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 2]
```diff
@@ -99,21 +99,6 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         )
     }
 
-    private val browsingModeListener = object : BrowsingModeListener {
-        override fun onBrowsingModeChange(newMode: BrowsingMode) {
-            themeManager.currentTheme = newMode
-        }
-    }
-
-    override fun applyOverrideConfiguration(overrideConfiguration: Configuration?) {
```
Comment (boek) [inline app/src/main/java/org/mozilla/fenix/HomeActivity.kt]: This is reverted from: https://github.com/mozilla-mobile/fenix/commit/1b7cb117cade9500e0130d7f3177b115ae7bb626#diff-44fc84568dc7e79b2cb8921c2135183a

**Comment by cadeyrn** [comment]
@boek: Since one commit mentions "Reverts For #7559" - Does this mean that the language selection will be broken again or do your changes include another fix for this bug?

**Comment by boek** [comment]
@cadeyrn I'm still working on a patch for that. But I think I might land this to bring the crash rate back down in Nightly

**Pull request merged**